dm/loader/lightning.go

```diff
@@ -325,14 +320,25 @@ func GetLightningConfig(globalCfg *lcfg.GlobalConfig, subtaskCfg *config.SubTask
 	cfg.App.RegionConcurrency = subtaskCfg.LoaderConfig.PoolSize
 	cfg.Routes = subtaskCfg.RouteRules
 
-	cfg.Checkpoint.Driver = lcfg.CheckpointDriverFile
-	var cpPath string
-	// l.cfg.LoaderConfig.Dir may be a s3 path, and Lightning supports checkpoint in s3, we can use storage.AdjustPath to adjust path both local and s3.
-	cpPath, err := storage.AdjustPath(subtaskCfg.LoaderConfig.Dir, string(filepath.Separator)+lightningCheckpointFileName)
-	if err != nil {
-		return nil, err
+	if subtaskCfg.ExtStorage != nil {
+		// if we use bucket as dumper storage, write lightning checkpoint to downstream DB to avoid bucket ratelimit
+		// it's the default path for dm on cloud
+		cfg.Checkpoint.Driver = lcfg.CheckpointDriverMySQL
+		connParams, err := connParamFromDBConfig(&subtaskCfg.To)
+		if err != nil {
+			return nil, err
+		}
+		cfg.Checkpoint.MySQLParam = connParams
```

Comment:
I forget if we can left this field empty, and let lightning automatically fill the value from lightning's target database

Reply:
I will check that logic.

Reply:
![image](https://user-images.githubusercontent.com/17703670/223899233-d86b3436-9228-490b-94d7-9969de6429ce.png)
Seems yes.

Reply:
Already manully test.